e2e/tests/functional/plugins/gauge/gauge.e2e.spec.js

```diff
@@ -133,4 +133,50 @@ test.describe('Gauge', () => {
 
     // TODO: Verify changes in the UI
   });
+
+  test('Gauge does not display NaN when data not available', async ({ page }) => {
+    // Create a Gauge
+    await createDomainObjectWithDefaults(page, {
+      type: 'Gauge',
+      name: 'Unnamed Gauge'
+    });
+
+    // Create a Sine Wave Generator in the Gauge with a loading delay
+    await page.getByRole('button', { name: ' Create ' }).click();
+    await page.getByRole('menuitem', { name: ' Sine Wave Generator' }).click();
+    await page
+      .getByRole('dialog')
+      .locator('div')
+      .filter({ hasText: 'Loading Delay (ms)' })
+      .nth(3)
+      .click();
+    await page.getByLabel('Loading Delay (ms)').fill('5000');
+    await page.locator('[aria-label="Save"]').click();
+    await page.getByLabel('Expand My Items folder').click();
+    await page
+      .getByRole('treeitem', { name: 'Expand Unnamed Gauge gauge  Unnamed Gauge' })
+      .locator('a')
+      .click();
+    const gaugeNoDataText = await page.locator('.js-dial-current-value tspan').textContent();
+    expect(gaugeNoDataText).toBe('--');
+  });
+
+  test('Gauge enforces composition policy', async ({ page }) => {
+    // Create a Gauge
+    await createDomainObjectWithDefaults(page, {
+      type: 'Gauge',
+      name: 'Unnamed Gauge'
+    });
+
+    // Try to create a Folder into the Gauge. Should be disallowed.
+    await page.getByRole('button', { name: ' Create ' }).click();
+    await page.getByRole('menuitem', { name: ' Folder' }).click();
```

Comment:
We can avoid the weird symbols in the locators by selection on a RegExp instead:

```suggestion
    await page.getByRole('button', { name: /Create/ }).click();
    await page.getByRole('menuitem', { name: /Folder/ }).click();
```

Reply:
Implemented.